ext/crates/fp/src/limb.rs

```diff
@@ -11,6 +11,133 @@ pub(crate) struct LimbBitIndexPair {
     pub(crate) bit_index: usize,
 }
 
+/// A struct that defines a range of entries in a slice of limbs. We number limbs starting from some
+/// other limb that might not be in the range. This is because we must be able to index a struct
+/// using indices for another ambient struct. For example, accessing a `Slice` using coordinates
+/// from an ambient `FpVector`.
+#[derive(Debug, Hash, PartialEq, Eq, Clone, Copy)]
+pub struct LimbLength<const P: u32> {
+    /// The index of the first entry relative to some other base limb.
+    pub(crate) start: usize,
+    /// The index of the last entry relative to some other base limb.
+    pub(crate) end: usize,
+    /// The total number of limbs in the range.
+    pub(crate) limbs: usize,
+    /// The total number of entries in the range.
+    pub(crate) logical: usize,
+    /// The index of the first entry in the first limb in the range. This is a number between 0
+    /// (inclusive) and [`entries_per_limb_const::<P>()`] (exclusive).
+    pub(crate) offset: usize,
+    /// The total number of limbs in the range, from the other base limb to the last limb of this
+    /// range.
+    pub(crate) total_limbs: usize,
+}
```

Comment:
This struct is `Copy` and quite large, which might be a source of performance impact.

Reply:
That's a good point, I'll experiment with that.

Reply:
Fixed in [986b6b6](https://github.com/SpectralSequences/sseq/pull/53/commits/986b6b6565f94eca0d172cd0aed87d29dbd59ca4) (mostly)